sound/soc/sof/intel/hda-loader.c

```diff
@@ -306,24 +306,28 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	if (i == HDA_FW_BOOT_ATTEMPTS) {
 		dev_err(sdev->dev, "error: dsp init failed after %d attempts with err: %d\n",
 			i, ret);
-
-		hda_dsp_dump(sdev, SOF_DBG_REGS | SOF_DBG_PCI | SOF_DBG_MBOX);
-
-		/* disable DSP */
-		snd_sof_dsp_update_bits(sdev, HDA_DSP_PP_BAR,
-					SOF_HDA_REG_PP_PPCTL,
-					SOF_HDA_PPCTL_GPROCEN, 0);
-		goto done;
+		goto err;
 	}
 
 	/* at this point DSP ROM has been initialized and
 	 * should be ready for code loading and firmware boot
 	 */
 	ret = cl_copy_fw(sdev, tag);
-	if (ret < 0)
+	if (ret < 0) {
 		dev_err(sdev->dev, "error: load fw failed ret: %d\n", ret);
-	else
-		dev_dbg(sdev->dev, "Firmware download successful, booting...\n");
+		goto err;
+	}
+
+	dev_dbg(sdev->dev, "Firmware download successful, booting...\n");
+	goto done;
+
+err:
+	hda_dsp_dump(sdev, SOF_DBG_REGS | SOF_DBG_PCI | SOF_DBG_MBOX);
+
+	/* disable DSP */
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_PP_BAR,
+				SOF_HDA_REG_PP_PPCTL,
+				SOF_HDA_PPCTL_GPROCEN, 0);
```

Comment:
going from here to done: looks odd, why would you get the stream tag at this point?

Reply:
@plbossart @keyonjie I've fixed this to disable the DSP after the stream cleanup.